src/main/kotlin/no/nav/familie/ef/sak/behandling/oppgaveforopprettelse/OppgaverForOpprettelseService.kt

```diff
@@ -0,0 +1,65 @@
+import no.nav.familie.ef.sak.behandling.domain.BehandlingType
+import no.nav.familie.ef.sak.infrastruktur.exception.feilHvisIkke
+import no.nav.familie.ef.sak.tilkjentytelse.TilkjentYtelseService
+import no.nav.familie.ef.sak.tilkjentytelse.domain.TilkjentYtelse
+import no.nav.familie.kontrakter.ef.iverksett.OppgaveForOpprettelseType
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+import java.util.UUID
+
+@Service
+class OppgaverForOpprettelseService(
+    private val oppgaverForOpprettelseRepository: OppgaverForOpprettelseRepository,
+    private val behandlingService: BehandlingService,
+    private val tilkjentYtelseService: TilkjentYtelseService,
+) {
+
+    @Transactional
+    fun opprettEllerErstattFremleggsoppgave(behandlingId: UUID, nyeOppgaver: List<OppgaveForOpprettelseType>) {
+        val oppgavetyperSomKanOpprettes = hentOppgavetyperSomKanOpprettes(behandlingId)
+        feilHvisIkke(oppgavetyperSomKanOpprettes.containsAll(nyeOppgaver)) {
+            "behandlingId=$behandlingId prøver å opprette $nyeOppgaver $oppgavetyperSomKanOpprettes"
+        }
+        when (this.oppgaverForOpprettelseRepository.existsById(behandlingId)) {
+            true -> this.oppgaverForOpprettelseRepository.update(OppgaverForOpprettelse(behandlingId, nyeOppgaver))
+            false -> this.oppgaverForOpprettelseRepository.insert(OppgaverForOpprettelse(behandlingId, nyeOppgaver))
+        }
+    }
+
+    fun hentOppgaverForOpprettelseEllerNull(behandlingId: UUID): OppgaverForOpprettelse? {
+        return oppgaverForOpprettelseRepository.findByIdOrNull(behandlingId)
+    }
+
+    fun hentOppgavetyperSomKanOpprettes(behandlingId: UUID): List<OppgaveForOpprettelseType> {
+        val behandling = behandlingService.hentBehandling(behandlingId)
+        val tilkjentYtelse = tilkjentYtelseService.hentForBehandlingEllerNull(behandlingId)
+        val kanOppretteInntektskontroll = kanOppretteOppgaveForInntektskontrollFremITid(behandling, tilkjentYtelse)
+
+        return if (kanOppretteInntektskontroll) listOf(OppgaveForOpprettelseType.INNTEKTSKONTROLL_1_ÅR_FREM_I_TID) else emptyList()
+    }
+
+    private fun kanOppretteOppgaveForInntektskontrollFremITid(
+        behandling: Behandling,
+        tilkjentYtelse: TilkjentYtelse?,
+    ): Boolean {
+        if (tilkjentYtelse == null) return false
+
+        val sisteAndel = tilkjentYtelse.andelerTilkjentYtelse.maxBy { it.stønadTom }
+        val sisteAndelMedBeløp = sisteAndel.beløp > 0
+        val sisteAndel1årFremITid = sisteAndel.stønadTom > LocalDate.now().plusYears(1)
+
+        return behandling.type == BehandlingType.FØRSTEGANGSBEHANDLING &&
+            sisteAndelMedBeløp &&
+            sisteAndel1årFremITid
+    }
+
+    fun slettOppgaverForOpprettelse(behandlingId: UUID) {
+        oppgaverForOpprettelseRepository.deleteById(behandlingId)
```

Comment:
Andre tilfeller hvor vi må vurdere å "slette" den;
- Ta av vent, med nullstilt vedtak
- Henleggelse av behandling (rydde opp?)
- Underkjent av beslutter? (det neste vedtaket må jo isåfall være gyldig for å kunne opprette fremleggsoppgave)